tf_encrypted/keras/engine/sequential.py

```diff
@@ -0,0 +1,104 @@
+"""Sequential model API."""
+import tf_encrypted as tfe
+from tf_encrypted.keras.engine.base_layer import Layer
+from tf_encrypted.keras.engine.input_layer import InputLayer, Input
+
+
+class Sequential(Layer):
+  """Model defined by a stack of layers in sequence.
+
+  TODO
+  """
+  def __init__(self, layers=None, name=None):
+    super(Sequential, self).__init__(name=name)
+
+    self._layers = []
+
+    # Add to the model any layers passed to the constructor.
+    if layers:
+      for layer in layers:
+        self.add(layer)
+
+  def add(self, layer):
+    """Adds a layer instance on top of the layer stack.
+    Arguments:
+        layer: layer instance.
+    Raises:
+        TypeError: If `layer` is not a layer instance.
+        ValueError: In case the `layer` argument does not
+            know its input shape.
+        ValueError: In case the `layer` argument has
+            multiple output tensors, or is already connected
+            somewhere else (forbidden in `Sequential` models).
+    """
+    if not isinstance(layer, Layer):
+      raise TypeError('The added layer must be '
+                      'an instance of class Layer. '
+                      'Found: ' + str(layer))
+    self.built = False
+    set_inputs = False
+    if not self._layers:
+      if isinstance(layer, InputLayer):
+        raise ValueError("Do not manually define an InputLayer in your "
+                         "tfe.keras.Sequential model.")
+
+      batch_shape = layer._batch_input_shape  # pylint: disable=protected-access
+
+      # Instantiate an input layer.
+      x = Input(
+          batch_shape=batch_shape,
+          name=layer.name + '_input')
+      # This will build the current layer
+      # and create the node connecting the current layer
+      # to the input layer we just created.
+      y = layer(x)
+
+      # If an input layer (placeholder) is available.
+      if isinstance(y, (tuple, list)):
+        raise ValueError('All layers in a Sequential model '
+                         'should have a single output tensor. '
+                         'For multi-output layers, '
+                         'use the functional API.')
+      self.outputs = [y]
+
+    elif self.outputs:
+      # If the model is being built continuously on top of an input layer:
+      # refresh its output.
+      output_tensor = layer(self.outputs[0])
+      if isinstance(output_tensor, list):
+        raise TypeError('All layers in a Sequential model '
+                        'should have a single output tensor. '
+                        'For multi-output layers, '
+                        'use the functional API.')
```

Comment:
do we have this yet? or is it part of not removing too much since we'll do this in the near future?

Reply:
yup, part of not removing too much